src/TreeMenu/index.tsx

```diff
@@ -63,11 +63,11 @@ class TreeMenu extends React.Component<TreeMenuProps, TreeMenuState> {
     }
   }
 
-  resetOpenNodes = (newOpenNodes?: string[]) => {
+  resetOpenNodes = (newOpenNodes?: string[], key?: string) => {
     const { initialOpenNodes } = this.props;
     const openNodes =
       (Array.isArray(newOpenNodes) && newOpenNodes) || initialOpenNodes || [];
-    this.setState({ openNodes, searchTerm: '' });
+    this.setState({ openNodes, searchTerm: '', activeKey: key || '', focusKey: '' });
```

Comment:
Nice that you also take into account `focusKey`. Why not also allow modifying `focusKey`? 🙂